monitor/sortedOracles.go

```diff
@@ -87,6 +89,50 @@ func (p sortedOraclesProcessor) ObserveState(opts *bind.CallOpts, stableTokenAdd
 	return nil
 }
 
+func (p sortedOraclesProcessor) ObserveMetric(opts *bind.CallOpts, stableTokenAddress common.Address) error {
+
+	isOldestReportExpired, _, err := p.sortedOracles.IsOldestReportExpired(opts, stableTokenAddress)
+	if err != nil {
+		return err
+	}
+	metrics.SortedOraclesIsOldestReportExpired.Set(utils.BoolToMetric(isOldestReportExpired))
+
+	numRates, err := p.sortedOracles.NumRates(opts, stableTokenAddress)
+	if err != nil {
+		return err
+	}
+	metrics.SortedOraclesNumRates.Set(float64(numRates.Uint64()))
+
+	medianRateNumerator, medianRateDenominator, err := p.sortedOracles.MedianRate(opts, stableTokenAddress)
+	if err != nil {
+		return err
+	}
+	medianRate := big.NewFloat(0)
+
+	if medianRateDenominator.Cmp(big.NewInt(0)) != 0 {
```

Comment:
should we use `FromFixed(medianRateNumerator)` and `FromFixed(medianRateDenominator)`

Reply:
maybe also want to move this to utils?